nncf/pruning/filter_pruning/algo.py

```diff
@@ -240,8 +243,11 @@ def _find_layerwise_pruning_rate(self, target_flops_pruning_rate):
 
     def set_pruning_rate(self, pruning_rate):
         # Pruning rate from scheduler can be flops pruning rate or percentage of params that should be pruned
+        same_pruning_rate = np.isclose(pruning_rate, self.pruning_rate)
         self.pruning_rate = pruning_rate
-        if not self.frozen:
+        if not self.frozen and not same_pruning_rate:
+            nncf_logger.info("Computing filter importances and masks,"
```

Comment:
I would keep only the first part with an ellipsis:
"Computing filter importances and masks..."